src/hooks/useUserAuth.ts

```diff
@@ -0,0 +1,17 @@
+import { setUser } from '../store/user/slice';
+import { useAppDispatch } from '../store/hooks';
+
+const useUserAuth = () => {
+  const dispatch = useAppDispatch();
+
+  const userJSON = localStorage.getItem('user');
+  const user = userJSON ? JSON.parse(userJSON) : false;
+  const isAuthenticated = Boolean(user.token);
+  dispatch(setUser(user));
```

Comment:
do we need to dispatch `setUser` every time? I don't think so.(at least in current implementation)
and of course it works, but let it do in useEffect

Reply:
fix: dispatched in useEffect